protostar/testing/test_collector.py

```diff
@@ -79,11 +72,12 @@ def fnmatch_any(name: str, pats: Iterable[str]) -> bool:
 TestSuiteInfoDict = Dict[TestSuitePath, TestSuiteInfo]
 
 
-class TestCollector:
-    @dataclass
-    class Config:
-        safe_collecting: bool = False
+class FunctionNameGetter(Protocol):
```

Comment:
TBH I don't understand the purpose of this protocol

Reply:
it's there to define an API for test collector name collecting functions.